### PR TITLE
Source URI Reloading backports

### PR DESCRIPTION
**Pull request by michalpristas**
Backport of multiple PRs related to config/source uri reloading

https://github.com/elastic/elastic-agent/pull/686
https://github.com/elastic/elastic-agent/pull/1252/

## How to test:
Standalone: 
- install
- run `./elastic-agent upgrade 8.9.10 --source-uri https://fake-uri.com

needs to fail with proper message

Managed
- install
- update policy to have custom source uri
- init upgrade
- see it fail with host not resolved

Succ scenario:
change uri to `http://localhost:8080` and start fileserver there
`python3 -m http.server`
directory needs to contain target asc, sha512 and archive files in location `beats/elastic-agent`

### CODE REVIEW

File: x-pack/elastic-agent/pkg/artifact/config_test.go [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+		require.Equal(t, tc.expectedInstallDirectory, cfg.InstallPath)
+		require.Equal(t, tc.expectedDropDirectory, cfg.DropPath)
+		require.Equal(t, tc.expectedTimeout, cfg.Timeout)
+
+		require.Equal(t, tc.expectedDisableProxy, cfg.Proxy.Disable)
+
+		if tc.expectedTLS {
+			require.NotNil(t, cfg.TLS)
+			require.Equal(t, tc.expectedTLSEnabled, *cfg.TLS.Enabled)
+			//require.Equal(t, tc.expectedFingerprint, cfg.TLS.CATrustedFingerprint)
```
Comment (pchila) [inline x-pack/elastic-agent/pkg/artifact/config_test.go:244]: leftover ?

**Review by pchila — COMMENTED** [review]
A couple of code style nitpicks but the major issue I have with this PR are the 2 binary files added to metricbeat (I suppose it's a wrong commit somewhere?)

File: x-pack/elastic-agent/pkg/artifact/config_test.go (hunk 1 above)
Comment (michalpristas) [reply x-pack/elastic-agent/pkg/artifact/config_test.go:244]: no it's backport, i'm copying code, this is just something not backported yet

**Comment by michalpristas** [comment]
@pchila i will ignore code style comments, as this is a code copy it would not be friendly for future backports.
those binary files i dont know how they got there, removing

**Review by pchila — APPROVED** [review]

**Pull request merged**